refactor: revised spring update policy (#7607)
camunda/camunda-docs: docs/reference/announcements-release-notes/release-policy.md

However, certain dependencies used by Camunda 8 may have a shorter maintenance policy than Camunda itself. Camunda may adopt a different update policy for these dependencies, as listed below.

> ### Spring Boot/Framework/Security Update Policies

Reviewer: Since this is already under a "Dependency maintenance policies" header, I suggest we use:

> Spring Boot/Framework/Security updates

Spring has a [different maintenance window](https://spring.io/projects/spring-boot#support) than Camunda for its open-source software (OSS) offering.

> Camunda addresses this with a dedicated Spring Boot/Framework/Security Update Policy for the Camunda Orchestration Cluster and for client artifacts.

Reviewer: ```suggestion
Camunda addresses this with Spring Boot/Framework/Security update policies for the Camunda Orchestration Cluster and client artifacts.
```

Reviewer: Also is "client artifacts" accurate? Later you used "client libraries".

Note: The version of the Spring Framework and Spring Security as well as other dependencies managed via Spring Boot (see [Managed Dependency Coordinates](https://docs.spring.io/spring-boot/appendix/dependency-versions/coordinates.html)) are covered by the same policy.

> #### Camunda Orchestration Cluster Spring Boot/Framework/Security Update Policy

Reviewer: This header is already under `Dependency maintenance policies > Spring Boot/Framework/Security updates`, so I would suggest we go with "Orchestration Cluster".

> The Camunda Orchestration Cluster is a Spring Boot application and leverages Spring Boot to execute fundamental functionality such as application configuration, REST infrastructure (including security), production ready features, etc.

Reviewer: ```suggestion
The Camunda Orchestration Cluster leverages Spring Boot to implement core functionality, like application configuration, REST infrastructure (including security), and other production-ready features.
```

> Orchestration Cluster patch releases may contain updates to newer Spring Boot minor versions, such that the **latest available patch release of Camunda Orchestration Cluster component uses a Spring version with an active OSS support window**.

Author: Thanks for the suggestion, I would revise it sligthly and actually flip the order of the sentences

>Camunda ensures that **the latest available patch releases make use of a Spring version with an active OSS support window**. Therefore, Orchestration Cluster patch releases **may contain updates to newer Spring Boot minor versions**.

wdyt?

> #### Camunda Client libraries Spring Boot Starters

Reviewer: This header is already under `Dependency maintenance policies > Spring Boot/Framework/Security updates`, so I would suggest we go with "Client libraries".

Camunda Clients or SDKs, such as the [Camunda Spring Boot Starter](../../apis-tools/camunda-spring-boot-starter/getting-started.md), that are meant to be included in third-party applications, may depend on a particular Spring Boot release.

> To not risk breaking backward compatibility in Camunda library patch releases, the Spring Boot version referenced is kept on the same minor release as used with the original Camunda minor release.

Reviewer: ```suggestion
To ensure backward compatibility, every Camunda library patch release references the same Spring Boot minor version as the one referenced in the Camunda library's latest minor release.
```

> However Camunda tests Camunda library patch releases for compatibility with newer Spring Boot minor releases, declaring compatibility with a newer Spring version with an active OSS support window at last one month before the OSS support window of the current latest supported Spring release ends. This is reflected in the Compatibility Matrix of these artifacts, e.g. the [Camunda Spring Boot Starter Compatibility Matrix](../../apis-tools/camunda-spring-boot-starter/getting-started.md#version-compatibility).

Reviewer: ```suggestion
However, Camunda tests newer Spring Boot minor releases for compatibility and declares compatible versions in the library's compatibility matrix. For an example, refer to the [Camunda Spring Boot Starter Compatibility Matrix](../../apis-tools/camunda-spring-boot-starter/getting-started.md#version-compatibility). 

If the library is compatible with a newer Spring version with an active OSS support window, Camunda declares compatibility at least one month before the support window ends.
```

The original sentence is quite wordy, so I tried to clean it up. Please let me know if I accurately captured everything in the revision.
